### ISSUE TITLE
CTRL_N in insert mode works like...

### ISSUE THREAD

**Issue by zuloloxi** [vim/vim#1788]
From VimGolf : Real Vim ninjas count every keystroke - do you?
# Start file
```
organize(cupboard, 3, 2);
prioritize(bureau, 8, 7);
realize(bannister, 4, 4);
moralize(railing, 3,9);
```

# End file
```
cupboard.organize(3, 2);
bureau.prioritize(8, 7);
bannister.realize(4, 4);
railing.moralize(3,9);
```
### Discussion
*Smylers2 user question:* Anybody able to explain why neither ```qai<C-N><C-N>.<Esc>3wdW<CR>q3@aZZ``` nor ```qai<C-N>.<Esc>3wdW<CR>q3@aZZ``` work? I'd expect <C-N> at the start of a line to insert the first word on the line (and so need <C-N><C-N> for the second one), so was surprised on the first line a single <C-N> inserting the second word — the further surprised to find on subsequent lines it does insert the first one. So qai<C-N>.<Esc>3wdW<CR>q3@aZZ does work, but uses up 4 keystrokes inserting then deleting the black line at the start. 
*udioica user:* That's a bug that's only triggered when using <C-N> on top of a word at the first byte of a file. I've run into that bug many times before.

**Comment by chrisbra** [comment]
Can you provide a readable reproducible example that shows your problem? Sorry, pasting some discussion here does not make much sense.

**Comment by zuloloxi** [comment]
Thanks for very speedy answer, to resume :
Changing the text from:
```
organize(cupboard, 3, 2);
prioritize(bureau, 8, 7);
realize(bannister, 4, 4);
moralize(railing, 3,9);
```
to
```
cupboard.organize(3, 2);
bureau.prioritize(8, 7);
bannister.realize(4, 4);
railing.moralize(3,9);
```
with one of macros :
 ```qai<C-N><C-N>.<Esc>3wdW<CR>q3@aZZ``` 
or 
```qai<C-N>.<Esc>3wdW<CR>q3@aZZ```
doesn't work

**Comment by zuloloxi** [comment]
After this commit https://github.com/zuloloxi/vim/commit/55b5755799f096597fff20e00036680afde5ff6c , the macro: 
```
qai<C-N>.<Esc>3wdW<CR>q3@aZZ
``` 
to change the text from :
```
organize(cupboard, 3, 2);
prioritize(bureau, 8, 7);
realize(bannister, 4, 4);
moralize(railing, 3,9);
```
to
```
cupboard.organize(3, 2);
bureau.prioritize(8, 7);
bannister.realize(4, 4);
railing.moralize(3,9);
``` 
works !

**Comment by chrisbra** [comment]
Hi, I could reproduce the issue and created a [fix](https://gist.github.com/chrisbra/231eb00e48caf163fd34278b73194225)

**Comment by zuloloxi** [comment]
Thanks a lot for the nice work ! Would be nice to use one CTRL-N -> and to start completion with next word on line (and not the current cursor word) .
Best regards !

**Comment by chrisbra** [comment]
Fixed by [v8.0.0669](https://github.com/vim/vim/releases/tag/v8.0.0669)

**Comment by zuloloxi** [comment]
Thanks, all the best !
